src/NHibernate/Dialect/Function/RoundFunction.cs

```diff
@@ -12,9 +12,20 @@ namespace NHibernate.Dialect.Function
 	[Serializable]
 	public class RoundEmulatingSingleParameterFunction : ISQLFunction
 	{
-		private static readonly ISQLFunction SingleParamRound = new SQLFunctionTemplate(null, "round(?1, 0)");
+		private readonly ISQLFunction _singleParamRound;
+		private readonly ISQLFunction _round;
+		private readonly string _name;
 
-		private static readonly ISQLFunction Round = new StandardSQLFunction("round");
+		/// <summary>
+		/// Constructs a <c>RoundEmulatingSingleParameterFunction</c>.
+		/// </summary>
+		/// <param name="name">The SQL name of the round function to call.</param>
+		public RoundEmulatingSingleParameterFunction(string name)
```

Comment:
Class is public, so you need to add a default constructor.

Reply:
For avoiding a breaking change? This class has not yet been published.

Reply:
Ok. Fine then.